### PR TITLE
#1734 rename "FileKey" to "DataKey"

### PR DESCRIPTION
**Pull request by chia7712**
### Checklist
- [x] [Review Contribution Guidelines](https://github.com/oharastream/ohara/blob/master/CONTRIBUTING.md)
- [x] Add Reviewers (see [Authors](https://github.com/oharastream/ohara#authors))
- [x] Add Assignees (of course, you are the assignee by default!)
- [x] DON'T set a Milestone
- [x] DON'T add any labels (PR labels will be added automatically by Jenkins)
- [x] Link to an issue(You can do so by using the required by keyword: required by #1734)
- [x] Add unit tests (or please explain why this PR is not worth having new tests)
- [ ] Pass QA (Sometimes you may encounter an existing flaky test which would obstruct you from getting QA passed. If the failed tests are unrelated, you can add a comment in the PR thread)

### CODE REVIEW

File: ohara-client/src/main/scala/com/island/ohara/client/configurator/v0/DataKey.scala [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+          name = noJsNull(json.asJsObject.fields)
+            .get(Data.NAME_KEY)
+            .map(_.convertTo[String])
+            .getOrElse(
+              throw DeserializationException(s"where is your ${Data.NAME_KEY} ???", fieldNames = List(Data.NAME_KEY))),
+        )
+        catch {
+          case e: Throwable => throw DeserializationException("failed to generate DataKey from input json", e)
+        }
+
```
Comment (saivirtue) [inline ohara-client/src/main/scala/com/island/ohara/client/configurator/v0/DataKey.scala]: Do we need to add this parser for read()? Are there auto checking for required field in `jsonFormat2[DataKey]` ?

File: ohara-client/src/main/scala/com/island/ohara/client/configurator/v0/DataKey.scala [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+package com.island.ohara.client.configurator.v0
+
+import spray.json.DefaultJsonProtocol._
+import spray.json.{DeserializationException, JsObject, JsString, JsValue, RootJsonFormat}
+
+/**
+  * we are about to live in a new world full of group and name. The pure string in the past is deprecated. Now, a whole
+  * new class called DataKey is used to destroy the old world and lead us to enter a beautiful new world.
+  *
+  * NOTED: In order to accept the newbie who don't have group, the default value to group could be Data.DEFAULT_GROUP.
```
Comment (saivirtue) [inline ohara-client/src/main/scala/com/island/ohara/client/configurator/v0/DataKey.scala]: ~Data.DEFAULT_GROUP~ Data.GROUP_DEFAULT :)

File: ohara-client/src/main/scala/com/island/ohara/client/configurator/v0/DataKey.scala (hunk 1 above)
Comment (chia7712) [reply ohara-client/src/main/scala/com/island/ohara/client/configurator/v0/DataKey.scala]: akka-json can't generate format for the case class which have companion object :(

Comment (saivirtue) [reply ohara-client/src/main/scala/com/island/ohara/client/configurator/v0/DataKey.scala]: So bad...

**Comment by chia7712** [comment]
retry

**Comment by chia7712** [comment]
retry

**Comment by chia7712** [comment]
retry quickly

**Comment by chia7712** [comment]
retry

**Pull request merged**